src/Agent.Listener/Configuration/AutoLogonRegistryManager.cs

```diff
@@ -0,0 +1,339 @@
+            FetchAutoLogonUserDetails(out string autoLogonUserName, out string autoLogonUserDomainName);
+            
+            if (autoLogonUserName != null
+                    && autoLogonUserDomainName != null
+                    && !domainName.Equals(autoLogonUserDomainName, StringComparison.CurrentCultureIgnoreCase)
+                    && !userName.Equals(autoLogonUserName, StringComparison.CurrentCultureIgnoreCase))
+            {
+                var terminal = HostContext.GetService<ITerminal>();
+                terminal.WriteLine(StringUtil.Loc("AutoLogonAlreadyEnabledWarning", userName));
+            }
+        }
+
+        private void FetchAutoLogonUserDetails(out string userName, out string domainName)
+        {
+            userName = null;
+            domainName = null;
+
+            var regValue = _registryManager.GetValue(RegistryHive.LocalMachine, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogon);
+            if (int.TryParse(regValue, out int autoLogonEnabled)
+                    && autoLogonEnabled == 1)
+            {
+                userName = _registryManager.GetValue(RegistryHive.LocalMachine, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogonUserName);
+                domainName = _registryManager.GetValue(RegistryHive.LocalMachine, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogonDomainName);
+            }
+        }
+    }
+
+    public class RegistryConstants
+    {
+        public const string BackupKeyPrefix = "VSTSAgentBackup_";
+
+        public struct SubKeys
+        {
+            public const string ScreenSaver = @"Control Panel\Desktop";
+            public const string ScreenSaverDomainPolicy = @"Software\Policies\Microsoft\Windows\Control Panel\Desktop";
+            public const string StartupProcess = @"SOFTWARE\Microsoft\Windows\CurrentVersion\Run";
+            public const string AutoLogon = @"SOFTWARE\Microsoft\Windows NT\CurrentVersion\Winlogon";
+            public const string ShutdownReasonDomainPolicy = @"SOFTWARE\Policies\Microsoft\Windows NT\Reliability";
+            public const string LegalNotice = @"SOFTWARE\Microsoft\Windows NT\CurrentVersion\Winlogon";
+        }
+
+        public struct ValueNames
+        {
+            public const string ScreenSaver = "ScreenSaveActive";
+            public const string AutoLogon = "AutoAdminLogon";
+            public const string AutoLogonUserName = "DefaultUserName";
+            public const string AutoLogonDomainName = "DefaultDomainName";
+            public const string AutoLogonCount = "AutoLogonCount";
+            public const string AutoLogonPassword = "DefaultPassword";
+            public const string StartupProcess = "VSTSAgent";
+            public const string ShutdownReason = "ShutdownReasonOn";
+            public const string ShutdownReasonUI = "ShutdownReasonUI";
+            public const string LegalNoticeCaption = "LegalNoticeCaption";
```

Comment:
For any non-obvious settings, can you add comments that explain the significance? AutoLogonCount, ShutdownReason, ShutdownReasonUI, LegalNoticeCaption, LegalNoticeText. Inline with the methods that set them might be a better place, since that would describe that part of the code inline.